specification/cdn/resource-manager/Microsoft.Cdn/stable/2019-04-15/cdn.json

```diff
@@ -3230,6 +3231,109 @@
         }
       }
     },
+    "DeliveryRuleUrlSigningAction": {
+      "description": "Defines the Url Signing action for the delivery rule.",
+      "x-ms-discriminator-value": "UrlSigning",
+      "allOf": [
+        {
+          "$ref": "#/definitions/DeliveryRuleAction"
+        }
+      ],
+      "required": [
+        "parameters"
+      ],
+      "properties": {
+        "parameters": {
+          "description": "Defines the parameters for the action.",
+          "$ref": "#/definitions/UrlSigningParameters"
+        }
+      }
+    },
+    "UrlSigningParameters": {
+      "description": "Defines the parameters for the Url Signing action.",
+      "required": [
+        "keys",
+        "hmacAlgorithm",
+        "queryStringParams",
+        "useIpInHash"
+      ],
+      "keys": {
+        "description": "Keys to be used for verifying the hash",
+        "type": "array",
+        "items": {
+          "$ref": "#/definitions/UrlSigningKey"
+        }
+      },
+      "hmacAlgorithm": {
+        "description": "Algorithm to use for URL signing",
+        "type": "string",
+        "enum": [
+          "SHA256",
+          "SHA1"
+        ]
+      },
+      "queryStringParams": {
+        "description": "Defines which query string parameters in the url to be considered for expiry, key id etc. ",
+        "type": "array",
+        "items": {
+          "$ref": "#/definitions/UrlSigningQueryStringParamIdentifier"
+        }
+      },
+      "useIpInHash": {
+        "description": "Indicates whether ip should be considered as part of the hash",
+        "type": "boolean"
+      }
+    },
+    "UrlSigningKey": {
+      "description": "Url signing key properties",
+      "required": [
+        "keyId",
+        "keyVaultName",
+        "keyName",
+        "keyVersion"
+      ],
+      "properties": {
+        "keyId": {
+          "description": "Logical reference to the key. Used in the URL to reference which key is used for producing the hash",
+          "type": "string"
+        },
+        "keyVaultName": {
+          "description": "Name of the key vault where the key is stored",
+          "type": "string"
+        },
+        "keyName": {
+          "description": "Name of the key",
+          "type": "string"
+        },
+        "keyVersion": {
+          "description": "Version of the key",
+          "type": "string"
+        }
+      }
+    },
+    "UrlSigningQueryStringParamIdentifier": {
+      "description": "Defines how to identify a query string parameter for a specific purpose e.g. expiry",
+      "required": [
+        "paramIndicator",
+        "paramName"
+      ],
+      "properties": {
+        "paramIndicator": {
+          "description": "Indicates the purpose of the query string parameter",
+          "type": "string",
+          "enum": [
+            "expiry",
```

Comment:
>expiry [](start = 13, length = 6)

change to expires?

Reply:
new PR>

---
In reply to: [302759997](https://github.com/Azure/azure-rest-api-specs/pull/6438#discussion_r302759997) [](ancestors = 302759997)